src/smart-escrow/SmartEscrow.sol

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @notice The error is thrown when an address is not set.
+error AddressIsZeroAddress();
+
+/// @notice The error is thrown when the start timestamp is zero.
+error StartTimeIsZero();
+
+/// @notice The error is thrown when the start timestamp is greater than the end timestamp.
+/// @param startTimestamp The provided start time of the contract
+/// @param endTimestamp The provided end time of the contract
+error StartTimeAfterEndTime(uint256 startTimestamp, uint256 endTimestamp);
+
+/// @notice The error is thrown when the vesting period is zero.
+error VestingPeriodIsZeroSeconds();
+
+/// @notice The error is thrown when the caller of the method is not the expected owner.
+/// @param caller The address of the caller
+/// @param owner The address of the owner
+error CallerIsNotOwner(address caller, address owner);
+
+/// @notice The error is thrown when the contract is terminated, when it should not be.
+error ContractIsTerminated();
+
+/// @notice The error is thrown when the contract is not terminated, when it should be.
+error ContractIsNotTerminated();
+
+/// @title SmartEscrow contract
+/// @notice Contract to handle payment of OP tokens over a period of vesting with
+///         the ability to terminate the contract.
+/// @notice This contract is inspired by OpenZeppelin's VestingWallet contract, but had 
+///         sufficiently different requirements to where inheriting did not make sense.
+contract SmartEscrow is Ownable2Step {
+    /// @notice Event emitted when the beneficiary owner is updated.
+    /// @param oldOwner The address of the old beneficiary owner
+    /// @param newOwner The address of the new beneficiary owner
+    event BeneficiaryOwnerUpdated(address indexed oldOwner, address indexed newOwner);
```

Comment:
official style guide is events go below state variables 
<img width="540" alt="image" src="https://github.com/base-org/contracts/assets/6678357/92280aee-2e8b-4028-8615-be8fd1fad3b9">
https://docs.soliditylang.org/en/latest/style-guide.html